src/neo/SmartContract/Iterators/StorageIterator.cs

```diff
@@ -7,30 +8,48 @@ namespace Neo.SmartContract.Iterators
     internal class StorageIterator : IIterator
     {
         private readonly IEnumerator<(StorageKey Key, StorageItem Value)> enumerator;
+        private readonly FindOptions options;
+        private readonly ReferenceCounter referenceCounter;
 
-        public StorageIterator(IEnumerator<(StorageKey, StorageItem)> enumerator)
+        public StorageIterator(IEnumerator<(StorageKey, StorageItem)> enumerator, FindOptions options, ReferenceCounter referenceCounter)
         {
             this.enumerator = enumerator;
+            this.options = options;
+            this.referenceCounter = referenceCounter;
         }
 
         public void Dispose()
         {
             enumerator.Dispose();
         }
 
-        public PrimitiveType Key()
-        {
-            return enumerator.Current.Key.Key;
-        }
-
         public bool Next()
         {
             return enumerator.MoveNext();
         }
 
         public StackItem Value()
         {
-            return enumerator.Current.Value.Value;
+            byte[] key = enumerator.Current.Key.Key;
+            byte[] value = enumerator.Current.Value.Value;
+
+            if (options.HasFlag(FindOptions.RemovePrefix))
+                key = key[1..];
+
+            StackItem item = options.HasFlag(FindOptions.DeserializeValues)
+                ? BinarySerializer.Deserialize(value, 1024, (uint)value.Length, referenceCounter)
+                : value;
+
+            if (options.HasFlag(FindOptions.PickField0))
```

Comment:
If has both flags? `PickField0` and `PickField1`?

Reply:
https://github.com/neo-project/neo/blob/b29e24e5b64db399aca52ca6172407846bb5e25c/src/neo/SmartContract/ApplicationEngine.Storage.cs#L67-L68